cypress-io/cypress
# feat: Added configuration parameter that controls the amount of connection attempts to the browser

### Additional details
Currently, there is a hardcoded limit of 62 attempts until a connection to the browser is considered fails. For my usecase, this timeout is sometimes to low and i can't change it.

I added a config value and made sure that it is used at the corresponding places when connections to browsers is attempted.

### Steps to test
1. Try to start a browser without changing the config. Everything should work just fine
2. Set connectRetryThreshold to an absurd low value (like 5). Connection should fail due to timeout

### How has the user experience changed?
In the default usecase, Experience did not change. However when users experience connection timeouts, User can adjust the config value to stabilize their tests.

### PR Tasks
- [ x] Have tests been added/updated?
- [ x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? (https://github.com/cypress-io/cypress-documentation/pull/5070)
- [x ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

###  Other

As this is my first PR for cypress, i am unsure if I did everything right. Should something be amiss, please contact me and I will try my best to solve the issues.
